### PR TITLE
AZP: build static like a separate stage

### PR DESCRIPTION
**Pull request by avildema**
move build with static library to separate stage in CI

### CODE REVIEW

**Review by hoopoepg — APPROVED** [review]

File: buildlib/pr/main.yml [hunk 1]
```diff
@@ -296,6 +282,46 @@ stages:
...
+              extra_tls: dc_mlx5 rc_mlx5 ud_mlx5 rc_verbs ud_verbs cma
+              run_tls: ib rc rc_v rc_x dc dc_x ud ud_v ud_x shm sm
+            ubuntu2004:
+              CONTAINER: ubuntu2004
+              extra_tls: ""
+              run_tls: ""
+            ubuntu1804:
+              CONTAINER: ubuntu1804
+              extra_tls: ""
+              run_tls: ""
```
Comment (yosefe) [inline buildlib/pr/main.yml:307]: @hoopoepg do we need to pass shared memory TLs here?

File: buildlib/tools/build_static.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
+#!/bin/bash -eExl
```
Comment (yosefe) [inline buildlib/tools/build_static.sh:1]: add copyright

File: buildlib/tools/build_static.sh [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+	# Set port number for hello_world applications
+	server_port=$((10000 + (1000 * EXECUTOR_NUMBER)))
+	server_port_arg="-p $server_port"
+
+	for tls in tcp $RUN_TLS; do
+		echo UCX_TLS=$tls
+		UCX_TLS=$tls ./ucp_hello_world_static ${server_port_arg} &
+		# allow server to start
+		sleep 3
```
Comment (yosefe) [inline buildlib/tools/build_static.sh:46]: sleep 10

File: buildlib/tools/build_static.sh [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+		UCX_TLS=$tls ./ucp_hello_world_static ${server_port_arg} -n localhost
```
Comment (yosefe) [inline buildlib/tools/build_static.sh:47]: also let's wait for server to exit

File: buildlib/tools/build_static.sh [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+	done
+
+	cd -
+
+	export LD_LIBRARY_PATH=$SAVE_LD_LIBRARY_PATH
+
+	$MAKEP uninstall
```
Comment (yosefe) [inline buildlib/tools/build_static.sh:54]: why needed?

File: buildlib/tools/build_static.sh [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+	amount=$1
+	shift
+	# cleanup build dir before the task
+	[ -n "${ucx_build_dir}" ] && rm -rf ${ucx_build_dir}/*
+
+	$@
+
+	echo "##vso[task.setprogress value=$PROGRESS;]Progress Indicator"
+	PROGRESS=$((PROGRESS+amount))
+}
```
Comment (yosefe) [inline buildlib/tools/build_static.sh:73]: can remove this wrapper?

File: buildlib/pr/main.yml (hunk 1 above)
Comment (hoopoepg) [reply buildlib/pr/main.yml:307]: yes: empty tls list means that build should complete, but testing should not run (as I remember some docker containers are not configured properly to run tests)

**Review by yosefe — APPROVED** [review]

File: buildlib/pr/main.yml [hunk 7]
```diff
@@ -219,10 +209,6 @@ stages:
               BUILD_ID: "$(Build.BuildId)-$(Build.BuildNumber)"
               long_test: $(long_test)
               test_static: $(test_static)
-              EXTRA_MODULES: $(extra_modules)
```
Comment (hoopoepg) [inline buildlib/pr/main.yml:222]: should not be removed

**Review by yosefe — APPROVED** [review]
@hoopoepg can you pls re-review as well?

**Review by hoopoepg — APPROVED** [review]
should be ok now

**Pull request merged**